rfcs/0013-redesign-assets-id-system.md

```diff
@@ -0,0 +1,255 @@
+# Redesign Assets System
+
+This RFC will serve to finalize our approach to refactoring assets-registry and 
+removing assets, currency-factory, and our dependence on hard-coded assets.
+
+## Context
+
+Previously, we planned to use currency-factory to reserve asset IDs. However,
+currency-factory itself was mostly difficult to maintain legacy code with an
+asset ID range system that was overly complex and not helpful to our current 
+tech stack.
+
+As it stands, we have removed most of the dependence on currency-factory from
+Picasso in favor of hard-coded asset IDs. These hard-coded IDs have allowed us
+to conduct an initial release using a simplistic storage migration to create
+initial pools for Pablo. However, with the dependence on these hard-coded asset
+IDs - we now must conduct additional, more complex, storage migrations to add 
+more Pablo pools and more tokens.
+
+Instead of depending on these hard-coded asset IDs, we should instead develop
+a dynamic ID system that makes it easy for developers to both interact with 
+assets within our consensus system and to retrieve information relevant to a 
+given asset.
+
+## Requirements
+
+* There exists a double mapping between multi-locations and local asset IDs
+  * Zero or One multi-locations ↔ One asset ID
+
+* Given a way to uniquely identify an asset, one should be able to retrieve 
+asset metadata (ticker-symbol, decimal precision, and ratio)
+
+* Support the traits from `frame_support::traits::tokens::fungibles`
+  * Specifically `MutateHold`
+
+* Ability to mint mintable assets and the inability to mint non-mintable assets
+
+* Support the existing local asset IDs we have hard-coded
+
+* Supply an Asset Transactor to our XCM Config
+
+# Other Solutions
+
+* [Acala](./0013/acala-analysis.md)
+* [Moonbeam](./0013/moonbeam-analysis.md)
+* [Parallel](./0013/parallel-finance-analysis.md)
+
+<!-- TODO: These are mostly done, but I will introduce them to the repo in 
+their own files -->
+
+# Design
```

Comment:
Would be nice to have some overview diagram of how the assets system components fit together